source/initial_conditions/adiabatic.cc

```diff
@@ -58,6 +52,21 @@ namespace aspect
                                                   :
                                                   adiabatic_surface_temperature;
 
+      // then, get the temperature at the top and bottom boundary of the model
+      // if no boundary temperature is prescribed simply use the adiabatic
+      const double T_surface = (&this->get_boundary_temperature() != 0)
+                               ?
+                                   this->get_boundary_temperature().minimal_temperature(
+                                       this->get_fixed_temperature_boundary_indicators())
+                               :
+                                   adiabatic_surface_temperature;
+      const double T_bottom = (&this->get_boundary_temperature() != 0)
+                              ?
+                                  this->get_boundary_temperature().maximal_temperature(
+                                      this->get_fixed_temperature_boundary_indicators())
+                              :
+                                  adiabatic_surface_temperature;
+
```

Comment:
I never liked the minimal_temperature() and maximal_temperature() anyway. It is hard to know these values (e.g. if you read in data) and you can actually prescribe them in useless ways in some plugins.
I would suggest to change the code to actually do what the comment suggests ;-) ... We could ask for the boundary_temperature at the surface and at the bottom, instead of using these strange minimal_temperature() functions. Then we would be one step closer to getting rid of those.